aptos-move/block-executor/src/executor.rs

```diff
@@ -143,6 +162,100 @@ where
         }
     }
 
+    fn execute_v2(
+        idx_to_execute: TxnIndex,
+        incarnation: Incarnation,
+        signature_verified_block: &TP,
+        last_input_output: &TxnLastInputOutput<T, E::Output, E::Error>,
+        versioned_cache: &MVHashMap<T::Key, T::Tag, T::Value, DelayedFieldID>,
+        executor: &E,
+        base_view: &S,
+        global_module_cache: &GlobalModuleCache<
+            ModuleId,
+            CompiledModule,
+            Module,
+            AptosModuleExtension,
+        >,
+        runtime_environment: &RuntimeEnvironment,
+        parallel_state: ParallelState<T>,
+        scheduler: &SchedulerV2,
+    ) -> Result<(), PanicError> {
+        let _timer = TASK_EXECUTE_SECONDS.start_timer();
+
+        // TODO(BlockSTMv2): proper integration w. execution pooling for performance.
+        let txn = signature_verified_block.get_txn(idx_to_execute);
+
+        let mut abort_manager = AbortManager::new(idx_to_execute, incarnation, scheduler);
+        let sync_view = LatestView::new(
+            base_view,
+            global_module_cache,
+            runtime_environment,
+            ViewState::Sync(parallel_state),
+            idx_to_execute,
+        );
+        let execution_result = executor.execute_transaction(&sync_view, txn, idx_to_execute);
+
+        let mut prev_modified_keys = last_input_output
+            .modified_keys(idx_to_execute, true)
+            .map_or_else(HashMap::new, |keys| keys.collect());
+        let mut read_set = sync_view.take_parallel_reads();
+        if read_set.is_incorrect_use() {
+            return Err(code_invariant_error(format!(
+                "Incorrect use detected in CapturedReads after executing txn = {idx_to_execute} incarnation = {incarnation}"
+            )));
+        }
+
+        let maybe_output =
+            Self::process_execution_result(&execution_result, &mut read_set, idx_to_execute)?;
+
+        // TODO: BlockSTMv2: use estimates for delayed field reads? (see V1 update on abort).
+        let mut resource_write_set = Vec::new();
+        if let Some(output) = maybe_output {
+            resource_write_set = output.resource_write_set();
+            for (key, value, maybe_layout) in resource_write_set.clone().into_iter() {
+                remove_from_previous_keys::<T>(&mut prev_modified_keys, &key, KeyKind::Resource)?;
+
+                abort_manager.invalidate_dependencies(versioned_cache.data().write_v2::<false>(
+                    key,
```

Comment:
When do we need to compare metadata?

Reply:
Metadata is groups only (though it is stored as resources), so I guess this comes in next PR